tools/NewHelpIndex.ps1

```diff
@@ -0,0 +1,146 @@
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+[CmdletBinding(
+    SupportsShouldProcess=$true
+)]
+param(
+    [Parameter(Mandatory = $false)]
+    [string] $Version,
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("Latest", "Stack")]
+    [string] $Target = "Latest",
+    [Parameter(Mandatory = $false)]
+    [string] $SourceBaseUri,
+    [Parameter(Mandatory = $false)]
+    [string] $EditBaseUri,
+    [Parameter(Mandatory = $false)]
+    [string] $BuildConfig = "Debug",
+    [Parameter(Mandatory = $false)]
+    [string] $OutputFile = "$PSCommandPath/../index.json"
+)
+
+if ([string]::isNullOrEmpty($Version))
+{
+    Import-LocalizedData -BindingVariable "AzureRMpsd1" -BaseDirectory $PSCommandPath/../AzureRM -FileName "AzureRM.psd1"
+    $Version = $AzureRMpsd1.ModuleVersion
```

Comment:
@maddieclayton for the places where we're using "default values" (`Version`, `SourceBaseUri`, `EditBaseUri`), we should write a message to the screen letting the user know what the values are being set to